### ISSUE TITLE
add a way to detect April 1st RFCs

### ISSUE THREAD

**Issue by reschke** [rfc-format/draft-iab-xml2rfc-v3-bis#30]
...because they get slightly different formatting of the pub date.

Or is the presence of "day=1" sufficient?

**Comment by reschke** [comment]
> Or is the presence of "day=1" sufficient?

Tried that, and it is not.

**Comment by reschke** [comment]
Proposed extension: <https://greenbytes.de/tech/webdav/rfc2629xslt.html#ext-rfc2629.date>

Example: <https://greenbytes.de/tech/webdav/rfc2616.html#RFC2324>

**Comment by reschke** [comment]
FWIW, the RFC reference files do not seem to include "day='1'" for April-1st RFCs.

**Comment by stpeter** [comment]
@reschke When you say that `day="1"` was not sufficient, what was lacking?

**Comment by reschke** [comment]
*If* we want to keep special treatment of April 1st RFCs, we need to define how a formatter is supposed to detect that something is one of those. (And then, the reference files should be updated to use that)

**Comment by jrlevine** [comment]
As far as I know, the only special formatting in April 1 RFCs is that they're dated "1 April 9999" rather than "April 9999" and day="1" does that. What am I missing?

**Comment by reschke** [comment]
What about RFCs that happen to be published on April 1, but are not April-1st RFCs? Impossible?

**Comment by jrlevine** [comment]
They say     <date month="04" year="2021"/>
same as any other RFC published in April.

**Comment by reschke** [comment]
Ok. So what we should document is that

1. for RFCs, no day-in-month should be given, except
2. for April-1st RFCs, in which case day="1" would be used.

That also means that xml2rfc should produce warnings/errors in case day is specified for a non-April-1st RFC (I haven't checked yet, but I'm pretty sure it does not)

**Comment by stpeter** [comment]
@reschke Agreed on points 1 and 2 as they relate to the XML of published RFCs. I'm not yet convinced that xml2rfc needs produce a warning or error, but that's likely a topic for the developers of xml2rfc.

**Comment by jrlevine** [comment]
It doesn't produce a warning, but this sounds like it belongs in the style guide, not xml2rfc. I would not want to add an `--is-april-fools-joke` switch.
